purchases/src/main/kotlin/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -177,6 +180,17 @@ internal open class DeviceCache(
         preferences.edit().putLong(customerInfoLastUpdatedCacheKey(appUserID), date.time).apply()
     }
 
+    @Synchronized
+    fun setStorefront(countryCode: String) {
+        verboseLog(BillingStrings.BILLING_STOREFRONT_CACHING.format(countryCode))
+        preferences.edit().putString(storefrontCacheKey, countryCode).apply()
+    }
+
+    @Synchronized
+    fun getStorefront(): String? {
+        return preferences.getString(storefrontCacheKey, null)
```

Comment:
I'd add a debug log if fetching this doesn't have a cache yet so we know if there are race conditions.

Reply:
Yeah, this should only happen the first time before the app has connecting to the billing client in theory, but it doesn't hurt 👍